modules/platforms/cpp/network/os/win/src/network/tcp_socket_client.cpp

```diff
@@ -28,6 +30,23 @@ namespace ignite
 {
     namespace network
     {
+        /**
+         * Shuffle addresses randomly.
+         * @param addrsIn Addresses.
+         * @return Randomly shuffled addresses.
+         */
+        std::vector<addrinfo*> ShuffleAddresses(addrinfo* addrsIn)
+        {
+            std::vector<addrinfo*> res;
+
+            for (addrinfo *it = addrsIn; it != NULL; it = it->ai_next)
+                res.push_back(it);
+
+            std::random_shuffle(res.begin(), res.end());
+
+            return res;
+        }
```

Comment:
This looks exactly like the code above for Linux, can we deduplicate?

Reply:
Well in theory, we can, using templates. The hard part here is that despite of the same name `addrinfo`s in Windows and Linux are two different classes, so we can not just make a simple function. I'll think if I can improve this part without over-complicating the code though.

Reply:
Fixed. Check out new version.

Reply:
Looks good, thanks!